### ISSUE TITLE
Random Files Generated

### ISSUE THREAD

**Issue by Manoj18Patel** [fabiocaccamo/django-freeze#89]
When i run the command : python manage.py generate_static_site
 then this files generated !
 what to do ??

![Screenshot 2023-07-04 180551](https://github.com/fabiocaccamo/django-freeze/assets/125271018/43de8813-c423-471e-9602-d83325bdebba)

## Funding
* You can sponsor this specific effort via a [Polar.sh](https://polar.sh) pledge below
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/fabiocaccamo/django-freeze/issues/89">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/fabiocaccamo/django-freeze/issues/89/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/fabiocaccamo/django-freeze/issues/89/pledge.svg">
</picture>
</a>

**Comment by fabiocaccamo** [comment]
@Manoj18Patel was your site navigable and working?
It hard to guess what happened only from the output, could you also provide some code?

**Comment by fabiocaccamo** [comment]
@Manoj18Patel any update?

**Comment by Manoj18Patel** [comment]
i am not able to give any code

**Comment by merwok** [comment]
Then we have to resort to divination to get info about your issue!

🔮 

Hm…  do you have a wrong `INSTALLED_APPS` with a string there interpreted as an iterable of characters?

**Comment by fabiocaccamo** [comment]
@Manoj18Patel I will be glad to re-open this issue in case you will provide a more useful feedback.

**Comment by anentropic** [comment]
I had exactly the same issue

I stuck a breakpoint in and the problem seems to be in this line:
https://github.com/fabiocaccamo/django-freeze/blob/0.10.0/freeze/writer.py#L101-L103

```python
                file_dst_path = file_src_path[
                    file_src_path.find(settings.FREEZE_STATIC_URL) :
                ]
```

`file_src_path.find(settings.FREEZE_STATIC_URL)` returns `-1`, so `file_dst_path` contains just the last letter of `file_src_path`, resulting in writing out a bunch of unreadable files with single-letter file names.

The logic seems wrong, there is no reason for `file_src_path` (location on disk, derived from `STATIC_ROOT`) to contain `FREEZE_STATIC_URL` (arbitrary url segment) except maybe by coincidence.

Here is a hacked version of this section of code that works for me now:
```python
            relative_static_root = pathlib.Path(root).relative_to(settings.FREEZE_STATIC_ROOT).as_posix()
            
            for file in files:
                file_src_path = os.path.join(root, file)
                file_dst_path = os.path.join(
                    settings.FREEZE_STATIC_URL.lstrip("/"),
                    file_src_path[file_src_path.find(relative_static_root):],
                )
```

**Comment by fabiocaccamo** [comment]
@anentropic thanks for having debugged this issue!

**Comment by fabiocaccamo** [comment]
@Manoj18Patel @anentropic if `file_src_path.find(settings.FREEZE_STATIC_URL)` returns `-1`, it means that your static file url doesn't start with `settings.FREEZE_STATIC_URL`, so probably there is something configured wrong. 

Could you provide the value of your `FREEZE_*` settings and also the value of `settings.STATIC_URL` ?

**Comment by anentropic** [comment]
It seems to me the mistake is searching for STATIC_URL in a path derived from STATIC_ROOT (`file_src_path`)

so e.g. I have `STATIC_ROOT = "/path/to/my/project/_deploy_static"` and `STATIC_URL = "/static/"`

and your code does:

```python
for root, _dirs, files in os.walk(settings.FREEZE_STATIC_ROOT):
    ...
    for file in files:
        file_src_path = os.path.join(root, file)
        file_dst_path = file_src_path[
            file_src_path.find(settings.FREEZE_STATIC_URL) :
        ]
```

the only way this can ever work currently is if someone happened to choose a STATIC_ROOT path which ends with the STATIC_URL

which might be quite common but is actually just a coincidence - the url is not derived from the static root path, they are unrelated variables

**Comment by fabiocaccamo** [comment]
@anentropic thanks for the explanation, got it!
I always set `STATIC_ROOT = "/path/to/my/project/static/"` that's why it worked well for me.

**Comment by fabiocaccamo** [comment]
@anentropic could you test if doing this fixes the issue please?
```python
for file in files:
    file_src_path = os.path.join(root, file)
    file_dst_path = file_src_path.lstrip(settings.FREEZE_STATIC_ROOT)
```

**Comment by merwok** [comment]
lstrip is to remove individual characters, maybe you mean removeprefix (added in a recent python version)?

**Comment by fabiocaccamo** [comment]
@merwok yes, thanks for the clarification!

**Comment by fabiocaccamo** [comment]
@merwok since this library still supports `Python 3.8`, `removeprefix` can't be used.

@anentropic could you test if doing this fixes the issue please?

```python
for file in files:
    file_src_path = os.path.join(root, file)
    file_dst_path = file_src_path
    if file_dst_path.startswith(settings.FREEZE_STATIC_ROOT):
        file_dst_path = file_dst_path[len(settings.FREEZE_STATIC_ROOT):]
    if file_dst_path.startswith("/"):
        file_dst_path = file_dst_path[1:]
```

**Comment by fabiocaccamo** [comment]
@anentropic any feedback?

**Comment by fabiocaccamo** [comment]
@Manoj18Patel any feedback?

**Comment by fabiocaccamo** [comment]
Fixed in `0.11.0` version.
